llama_index/objects/base_node_mapping.py

```diff
@@ -0,0 +1,68 @@
+"""Base object types."""
+
+from typing import TypeVar, Generic, Sequence, Type, Any
+from abc import abstractmethod
+from llama_index.data_structs.node import Node
+from llama_index.indices.base import BaseIndex
+from llama_index.indices.vector_store.base import VectorStoreIndex
+from llama_index.tools.types import BaseTool
+
+OT = TypeVar("OT")
+
+
+class BaseObjectNodeMapping(Generic[OT]):
+    """Base object node mapping."""
+
+    @classmethod
+    @abstractmethod
+    def from_objects(
```

Comment:
Do you think there should be an `as_object_index()` method too? Or maybe that's getting too circular haha

Reply:
+1 since we already have the objects and the self NodeMapping. The user can pass in their own Index type or we can default to Vector Index.

Reply:
Hm since the entry point is for the user to define an ObjectIndex as opposed to initializing a BaseNodeObjectMapping it would look like this:

```python
obj_index = ObjectIndex.from_objects(
    [list_tool, vector_tool], 
    VectorStoreIndex,
    SimpleToolNodeMapping
)
```

So IMO not a huge need to implement `as_object_index` atm